src/core/tenant-gate.ts

```diff
@@ -15,4 +15,14 @@ export class AllowAllTenantGate implements TenantGate {
   public async isTenant(_did: string): Promise<boolean> {
     return true;
   }
-}
\ No newline at end of file
+}
+/**
+ * A tenant gate that returns the value of externalTenantCheck, which requires a custom tenant checking function.
+ */
+export class ExternalTenantGate implements TenantGate {
+  constructor(private externalTenantCheck: Function) {}
```

Comment:
Not seeing a huge win here because the user will still need to implement `externalTenantCheck` somewhere. I see this as a convenience improvement and willing to approve it if `Function` is replaced with actual method signature, else it is arguably even less obvious (and error-prone) what is expected of the class.

Also agree with @diehuxx that adding explicit public facing documentation will help with understanding the intent and usage of tenant gate.

Reply:
Removed ExternalTenantGate.